src/core/StelPainter.cpp

```diff
@@ -2400,6 +2400,103 @@ void StelPainter::enableClientStates(bool vertex, bool texture, bool color, bool
 	normalArray.enabled = normal;
 }
 
+void StelPainter::drawFixedColorWideLineStripAsQuads(const ArrayDesc& vertexArray, int count, int offset, const Mat4f& projMat)
+{
+	GLint viewport[4] = {};
+	glGetIntegerv(GL_VIEWPORT, viewport);
+	const auto viewportSize = Vec2f(viewport[2], viewport[3]);
```

Comment:
for me, when reading code, I find the "auto" type specifier cumbersome, especially for actually simple data types. It always requires a second mental translation from the right side. It may be helpful in some situations to reduce complex casting syntax.

Reply:
Generally, it serves several purposes, one of which is DRY (and this one in particular lets avoid annoying situations when you forget to update one of the sides when changing types). Another is just that it doesn't really matter: the name is descriptive, and the type is "whatever is on the RHS, it's not important for reading".

Reply:
In cases when one might lose track of types or get hard to understand compiler errors I indeed use explicit types, like [around line 2440](https://github.com/Stellarium/stellarium/pull/3187/files#diff-824dbcd0573315173d549be8b09852aca7c49fbc51fb70efc1031fd583466465R2440).

Reply:
"viewportSize" may be reasonably descriptive, but "v0axy" (below) certainly is not. In QtCreator, type mismatch is nicely highlighted by clangd.

Reply:
Regarding `v0axy` I agree.
As for Clang model in QtCreator, it's a piece of crap that can't really parse lots of C and C++ code in my experience, and additionally, uses loads of RAM (or at least used to do so).

Reply:
Clang model in Qt Creator 10 is unusable

Reply:
I see a few exceptional cases where it fails in the Stars area, but in general works pretty well IMHO. It may use lots of RAM, but I have more than enough for that. It finds type mismatches, unused headers and many more useful things. Meanwhile it even gives git info about who last edited the line. As always, YMMV, use what you like best. On Raspi I use emacs...